contracts/clients-tests/BlsProvider.test.ts

```diff
@@ -0,0 +1,516 @@
+    expect(transactionReceipt.cumulativeGasUsed).to.equal(parseEther("0"));
+    expect(transactionReceipt.effectiveGasPrice).to.equal(parseEther("0"));
+  });
+
+  it("should retrieve a transaction receipt given a valid hash", async () => {
+    // Arrange
+    const recipient = signers[1].address;
+    const transactionResponse = await blsSigner.sendTransaction({
+      to: recipient,
+      value: parseEther("1"),
+    });
+
+    // Act
+    const transactionReceipt = await blsProvider.getTransactionReceipt(
+      transactionResponse.hash,
+    );
+
+    // Assert
+    // TODO: bls-wallet #412 Update values returned in bundle receipt to more closely match ethers transaction response
+    expect(transactionReceipt).to.be.an("object").that.deep.includes({
+      to: "0x",
+      from: "0x",
+      contractAddress: "0x",
+      // gasUsed: BigNumber.from("0"),
+      logsBloom: "",
+      logs: [],
+      confirmations: transactionResponse.confirmations,
+      // cumulativeGasUsed: parseEther("0"),
+      // effectiveGasPrice: parseEther("0"),
+      byzantium: false,
+      type: 2,
+    });
+
+    expect(transactionReceipt).to.include.keys(
+      "transactionIndex",
+      "blockHash",
+      "transactionHash",
+      "blockNumber",
+    );
+
+    // TODO: parseEther() doesn't work when asserting this way ^.
+    // Expects BigNumber { value: "0" } from transactionReceipt,
+    // but this returns { _hex: '0x00', _isBigNumber: true }.
+    expect(transactionReceipt.gasUsed).to.equal(parseEther("0"));
+    expect(transactionReceipt.cumulativeGasUsed).to.equal(parseEther("0"));
+    expect(transactionReceipt.effectiveGasPrice).to.equal(parseEther("0"));
+  });
+
+  it("gets a transaction given a valid transaction hash", async () => {
+    // Arrange
+    const recipient = signers[1].address;
+    const transactionAmount = parseEther("1");
+    const transactionRequest = {
+      to: recipient,
+      value: transactionAmount,
+    };
+
+    const expectedTransactionResponse = await blsSigner.sendTransaction(
+      transactionRequest,
+    );
+    const transactionReceipt = await expectedTransactionResponse.wait();
+
+    // Act
+    const transactionResponse = await blsProvider.getTransaction(
+      transactionReceipt.transactionHash,
+    );
+
+    // Assert
+    // TODO: bls-wallet #412 Update values returned in bundle receipt to more closely match ethers transaction response
+    expect(transactionResponse).to.be.an("object").that.deep.includes({
+      hash: transactionReceipt.transactionHash,
+      to: verificationGateway,
+      // TODO: Why is this different to signer wallet address? Uncomment logs below and run test to see problem.
+      // Will investigate this again when I look at bls-wallet #412.
+      // console.log("blsSigner wallet.address", blsSigner.wallet.address);
+      // console.log("Signer from:            ", expectedTransactionResponse.from);
+      from: "0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC",
```

Comment:
This may be the address of the aggregator that submitted the EOA txn w/ the bundle

Reply:
Turns out it's the third hardhat account. I haven't used `signers[2]` anywhere in my code.

```ts
Account #2: 0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC (10000 ETH)
Private Key: 0x5de4111afa1a4b94908f83103eb1f1706367c2e68ca870fc3fb9a804cdab365a
```

Could `0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC` be the address of the aggregator then?

Reply:
TODO @JohnGuilding @jacque006 Determine if this from address should be agg address or wallet address, will link here